clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/data/Keys.groovy

```diff
@@ -98,8 +100,19 @@ class Keys {
     "${AWS}:${Namespace.LAUNCH_CONFIGS}:${account}:${region}:${launchConfigName}"
   }
 
-  static String getLoadBalancerKey(String loadBalancerName, String account, String region, String vpcId) {
-    "${AWS}:${Namespace.LOAD_BALANCERS}:${account}:${region}:${loadBalancerName}${vpcId ? ':' + vpcId : ''}"
+  static String getLoadBalancerKey(String loadBalancerName, String account, String region, String vpcId, String type) {
+    String key = "${AWS}:${Namespace.LOAD_BALANCERS}:${account}:${region}:${loadBalancerName}"
+    if (vpcId) {
+      key += ":${vpcId}"
+    }
+    String loadBalancerType = type ?: 'classic'
+    if (loadBalancerType != 'classic') {
+      if (!vpcId) {
```

Comment:
probably safer than anything but it looks like ALB's are vpc-only so we should always have a `vpcId`?